python/parquet_tools/common/iofiles.py

```diff
@@ -0,0 +1,316 @@
+        xdir = ''
+        if tmplt[0:1] == '=':
+            n = tmplt.rfind('/')
+            if n < 0:
+                raise FatalError(3, 'directory specification in template',
+                                 stmp)
+            xdir = tmplt[1:n+1]
+            if n < len(tmplt)-1:
+                tmplt = tmplt[n+1:]
+            else:
+                tmplt = ''
+
+        # Get suffixes
+        #
+        dsfx = ''
+        asfx = ''
+        if tmplt:
+            if tmplt[0:1] == '-' and len(tmplt) > 1:
+                n = tmplt.find('+')
+                if n < 0:
+                    dsfx = tmplt[1:]
+                    tmplt = ''
+                else:
+                    dsfx = tmplt[1:n]
+                    tmplt = tmplt[n:]
+            if tmplt[0:1] == '+' and len(tmplt) > 1:
+                asfx = tmplt[1:]
+            else:
+                if tmplt:
+                    raise ParmError(3, 'suffix specification in template',
+                                    stmp)
+
+        # Apply template to generate output file names
+        #
+        for infile in IOFiles.fIN:
+            if xdir:
+                n = infile.rfind('/')
+                if n >= 0:
+                    infile = xdir+infile[n+1:]
+                else:
+                    infile = xdir+infile
+
+            if dsfx and infile.endswith(dsfx):
+                infile = infile[0:len(infile)-len(dsfx)]
+
+            if asfx:
+                infile += asfx
+
+            IOFiles.fOUT.append(infile)
+
+    # *****************************************************************************
+    # *                            _ f i l e _ o k                            *
+    # *****************************************************************************
+    @classmethod
+    def _file_ok(cls, infile):
+        """Verify that the given file exists and is an actual file.
+
+        Parameters
+        ----------
+        infile : 'string'
+            The input file path.
+
+        Raises
+        ------
+        ParmError
+
+        Returns
+        -------
+        infile : 'string'
+        """
+
+        if not os.path.exists(infile):
+            raise ParmError(6, 'Input file', infile, 'not found')
+        if not os.path.isfile(infile):
+            raise ParmError(6, 'Input file', infile, 'is not a file')
+        return infile
+
+    # *************************************************************************
+    # *                         g e t _ i n f i l e s                         *
+    # *************************************************************************
+    @classmethod
+    def get_infiles(cls, infile):
+        """Get all input files. The files are placed in the fIN vector.
+
+        Parameters
+        ----------
+        infile : 'string'
+            The command line input file specification.
+
+        Raises
+        ------
+        FatalError
+
+        Returns
+        -------
+        result : 'bool'
```

Comment:
don't name the return var. just say the type.

Reply:
Well, I will have to disagree on this one. The documentation specifically prohibits using this shorthand. See https://developer.lsst.io/python/numpydoc.html#py-docstring-returns this, as far as I can tell, is what LSST recommends using as a guide.

Reply:
Yes, you are meant to name the return var.